funcx_endpoint/funcx_endpoint/endpoint/rabbit_mq/task_queue_subscriber.py

```diff
@@ -372,13 +373,19 @@ def event_watcher(self):
                 self._watcher_poll_period, self.event_watcher
             )
 
+    def handle_sigterm(self, sig_num, curr_stack_frame):
+        # let the parent process do all the cleanup
+        # (it seems setting kill_event here can cause problems)
+        logger.warning("Received SIGTERM")
```

Comment:
I'm puzzled why this happens, but when I try to set `self.kill_event` here it seems to cause the setting of this exact same `multiprocessing.Event` in the main thread to block indefinitely. I don't think this should be the case, I should be able to set this event anywhere I want as many times as I want, if I understand correctly.

Regardless, giving up control here to the main process works perfectly fine and it shuts down this `TaskQueueSubscriber` process gracefully. It just bothers me a little to leave an empty SIGTERM handler here

Reply:
Deadlocks in multicomputing are almost always due to some variant of two or more execution threads acquiring resources in different orders.  (Proc 1 gets A waits for B, while Proc 2 gets B and waits for A).  Since `kill_event` is the culprit, that makes me think that it's not what it purports to be.  Where does `kill_event` come from?  Map that to it's source, and I'll bet you find a bug or a better understanding.

Reply:
> It just bothers me a little to leave an empty SIGTERM handler here

Then don't have an empty signal handler.  The point of this hunk was, I assume, to ignore the SIGTERM event?  In that case, it's sufficient to unset the _default_ SIGTERM signal handler:

```python
def run(self):
    ...
    signal.signal(signal.SIGTERM, self.SIG_IGN)
    ...
```

Reply:
Yup, I think it's easiest to just ignore the SIGTERM here